qiskit/circuit/library/generalized_gates/unitary.py

```diff
@@ -79,12 +85,12 @@ def __init__(self, data, label=None):
         data = numpy.array(data, dtype=complex)
         # Check input is unitary
         if not is_unitary_matrix(data):
-            raise ExtensionError("Input matrix is not unitary.")
+            raise ValueError("Input matrix is not unitary.")
```

Comment:
This feels like an error people could conceivably have been catching as part of a validation workflow.  During the deprecation period, we might have wanted to use a custom error type that derived both `ExtensionError` and `ValueError`, and only switched the documentation over to saying it raised `ValueError`.

Reply:
Yeah, I pointed this out in https://github.com/Qiskit/qiskit/pull/10725#discussion_r1308772310 as I wasn't sure whether we should take this precaution 🤔